Add separate author and committer support

elianddb opened the pull request:
Dolt now supports separate author (original intent) and committer (actual writer) identity for Dolt commits! Expect `commit`, `cherry-pick`, `revert` and `merge` operations (including SQL procedures) to pick up different author and committers automatically. The `dolt_log`, `dolt_commits`, `dolt_diff`, `dolt_column_diff`, `dolt_branches`, `dolt_remote_branches` system tables and respective table functions have been updated with author columns too.

> ⚠️ The committer columns in system tables or functions now show the committer identity instead of author. In old Dolt versions this was the result of no committer identity existing in storage, but has since been updated. We maintain backward-compatibility with this old format by duplicating the author identity into the committer identity. At serialization, however, the committer is only stored if the name or email is different from the author identity, allowing us to maintain the same commit hash history.
> ⚠️ We've appended the author columns under to end of each table to prevent issues with older Dolt clients. If you have a workflow that is negatively affected, we provide the `dolt_log_committer_only` system variable to remove the author columns for `dolt_log` and its table function.

New environment variables for committer identity are now available: `DOLT_COMMITTER_NAME`, `DOLT_COMMITTER_EMAIL`, `DOLT_COMMITTER_DATE` which take inspiration from their [`git` counterparts](https://git-scm.com/book/en/v2/Git-Internals-Environment-Variables#:~:text=GIT_AUTHOR_NAME%20is%20the%20human,in%20the%20%E2%80%9Ccommitter%E2%80%9D%20field.).
> ⚠️ Environment variables (not SQL system variables) are only interpretable by the Dolt client. MySQL or alternative SQL clients will not pick these Dolt-specific environment variables up automatically in your terminal.

```bash
export DOLT_COMMITTER_NAME="Committer User"
export DOLT_COMMITTER_EMAIL="committer@example.com"
export DOLT_COMMITTER_DATE="2025-01-01T12:00:00Z"
dolt commit -m "Commit with custom committer date"
```
```sql
+----------------------------------+----------------+-----------------------+---------------------+--------------------------------------+--------------+-----------+------------------+---------------------+
| commit_hash                      | committer      | email                 | date                | message                              | commit_order | author    | author_email     | author_date         |
+----------------------------------+----------------+-----------------------+---------------------+--------------------------------------+--------------+-----------+------------------+---------------------+
| tdf81ch4ug4c9uc2p78rls1iihj95unn | Committer User | committer@example.com | 2025-01-01 12:00:00 | Commit with custom committer date    | 4            | Test User | test@example.com | 2025-11-19 10:15:47 |
| 0gonkh11sf1esld53b66jdpbu78sdq6i | Committer User | committer@example.com | 2025-11-19 10:15:43 | Add Charlie with different committer | 3            | Test User | test@example.com | 2025-11-19 10:15:43 |
+----------------------------------+----------------+-----------------------+---------------------+--------------------------------------+--------------+-----------+------------------+---------------------+
```

Since non-Dolt clients cannot interpret these environment variables, we've also added the system variables `dolt_committer_name`, `dolt_committer_email`, `dolt_committer_date`, `dolt_committer_author`, `dolt_committer_email`, `dolt_committer_date`. They enable committer and author identity to travel over the wire.

- New environment variables for committer identity: `DOLT_COMMITTER_NAME`, `DOLT_COMMITTER_EMAIL`, and `DOLT_COMMITTER_DATE` 
- New session variables for author and committer identity: `dolt_committer_name`, `dolt_committer_email`, `dolt_committer_date`, `dolt_committer_author`, `dolt_committer_email`, `dolt_committer_date`
- Commit related operations now honor the committer identity system variables over the wire: `merge`, `cherry-pick`, `commit`, `revert`, `DOLT_TAG()` etc.
- `dolt` clients seed session identity variables from the `dolt config` on loopback connections, or the SQL client identity for non-loopback
- Add `dolt_log_committer_only` session variable to control if author columns are shown for `dolt_log` and `dolt_log()`
- Append author columns to `dolt_diff`, `dolt_column_diff`, `dolt_branches`, `dolt_remote_branches`, `dolt_history_*` system table, views and table functions
- Signed commits use a `V2` payload that appends `CommitterName`, `CommitterEmail`, and `CommitterDate` after the author fields, using the prior payload as a prefix for forward compatibility
- Commit flatbuffer gained `committer_name` and `committer_email` fields, written only when they differ from author
- `CommitDate` struct introduced to explicitly resolve author and committer timestamps at serialization unless otherwise specified by version control operations
Fix dolthub/dolt#1374
Blocks dolthub/doltgresql#2039
Blocks dolthub/docs#2826

elianddb committed:
separate author and committer identity in commit metadata

# Conflicts:
#	.github/workflows/ci-compatibility-tests.yaml
#	integration-tests/compatibility/runner.sh
#	integration-tests/compatibility/test_files/backward_compatible_versions.txt
#	integration-tests/compatibility/test_files/forward_compatible_versions.txt

# Conflicts:
#	go/libraries/doltcore/sqle/dtablefunctions/dolt_log.go
#	go/libraries/doltcore/sqle/dtables/log_table.go

coffeegoddd commented:
@elianddb __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>9308b60</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>9308b60</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@coffeegoddd __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>ae619d4</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>ae619d4</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

macneale4 commented on go/store/datas/commit_meta.go:
```diff
@@ -97,30 +102,54 @@ func init() {
+// If committer info is empty, defaults to author info.
+func NewCommitMetaWithAuthorCommitter(authorName, authorEmail, desc string, ats time.Time, committerName, committerEmail string, cts *time.Time) (*CommitMeta, error) {
+	an := strings.TrimSpace(authorName)
+	ae := strings.TrimSpace(authorEmail)
```
I hate short var names, despite golang communities obsession. And in this case, you don't want to use an untrimmed version of these values. So just over write them:

authorName = strings.TrimSpace(authorName)

desc = strings.TrimSpace(desc)

etc.

macneale4 commented on go/store/datas/commit_meta.go:
```diff
@@ -97,30 +102,54 @@ func init() {
-	committerDateMillis := uint64(CommitterDate().UnixMilli())
-	authorDateMillis := userTS.UnixMilli()
+	if cn == "" {
+		cn = an
```
Comoros is now Andorra?? What? Give me words I can read pls.

macneale4 commented on go/store/datas/commit_meta.go:
```diff
@@ -97,30 +102,54 @@ func init() {
+		cn = an
+	}
+	if ce == "" {
+		ce = ae
```
Sri Lanka = United Arab Emirates? Huh?

macneale4 commented on go/store/datas/commit_meta.go:
```diff
@@ -174,13 +203,18 @@ func CommitMetaFromNomsSt(st types.Struct) (*CommitMeta, error) {
+		Timestamp:      &ts,
+		UserTimestamp:  int64(userTS.(types.Int)),
+		CommitterName:  name,
+		CommitterEmail: email,
```
Is there no alternate committer in this context?

macneale4 commented on integration-tests/compatibility/test_files/bats/sql-server.bats:
There is a lot of copied code in here. Possible to put all the bats helper code in one place, or load based on a relative path?

coffeegoddd commented:
@elianddb __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>9398650</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>9398650</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@elianddb __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>2c224ad</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>2c224ad</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@elianddb __DOLT__
 <table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>847af37</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>847af37</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@elianddb __DOLT__
 |      read_tests      | from_latency | to_latency | percent_change |
|----------------------|--------------|------------|----------------|
| covering_index_scan  | 0.55         | 0.55       | 0.0            |
| groupby_scan         | 9.91         | 9.91       | 0.0            |
| index_join           | 1.82         | 1.82       | 0.0            |
| index_join_scan      | 1.37         | 1.34       | -2.19          |
| index_scan           | 21.89        | 21.89      | 0.0            |
| oltp_point_select    | 0.27         | 0.27       | 0.0            |
| oltp_read_only       | 5.37         | 5.37       | 0.0            |
| select_random_points | 0.54         | 0.54       | 0.0            |
| select_random_ranges | 0.56         | 0.55       | -1.79          |
| table_scan           | 22.28        | 22.28      | 0.0            |
| types_table_scan     | 66.84        | 66.84      | 0.0            |


|      write_tests      | from_latency | to_latency | percent_change |
|-----------------------|--------------|------------|----------------|
| oltp_delete_insert    | 6.32         | 6.32       | 0.0            |
| oltp_insert           | 3.13         | 3.07       | -1.92          |
| oltp_read_write       | 11.45        | 11.45      | 0.0            |
| oltp_update_index     | 3.19         | 3.25       | 1.88           |
| oltp_update_non_index | 3.19         | 3.13       | -1.88          |
| oltp_write_only       | 6.09         | 6.09       | 0.0            |
| types_delete_insert   | 6.91         | 6.91       | 0.0            |

coffeegoddd commented:
<!-- go-run-output -->
@elianddb __DOLT__
|      read_tests      | from_latency | to_latency | percent_change |
|----------------------|--------------|------------|----------------|
| covering_index_scan  | 0.54         | 0.55       | 1.85           |
| groupby_scan         | 9.91         | 11.24      | 13.42          |
| index_join           | 1.82         | 1.89       | 3.85           |
| index_join_scan      | 1.32         | 1.32       | 0.0            |
| index_scan           | 21.11        | 21.5       | 1.85           |
| oltp_point_select    | 0.25         | 0.25       | 0.0            |
| oltp_read_only       | 5.0          | 5.0        | 0.0            |
| select_random_points | 0.5          | 0.5        | 0.0            |
| select_random_ranges | 0.56         | 0.56       | 0.0            |
| table_scan           | 21.11        | 21.5       | 1.85           |
| types_table_scan     | 65.65        | 66.84      | 1.81           |


|      write_tests      | from_latency | to_latency | percent_change |
|-----------------------|--------------|------------|----------------|
| oltp_delete_insert    | 6.32         | 6.32       | 0.0            |
| oltp_insert           | 3.07         | 3.07       | 0.0            |
| oltp_read_write       | 11.04        | 11.04      | 0.0            |
| oltp_update_index     | 3.19         | 3.19       | 0.0            |
| oltp_update_non_index | 3.02         | 3.02       | 0.0            |
| oltp_write_only       | 5.99         | 5.99       | 0.0            |
| types_delete_insert   | 6.91         | 6.79       | -1.74          |

coffeegoddd commented:
<!-- go-run-output -->
@elianddb __DOLT__
|      test_name      | from_latency_p95 | to_latency_p95 | percent_change |
|---------------------|------------------|----------------|----------------|
| tpcc-scale-factor-1 | 53.85            | 54.83          | 1.82           |


|      test_name      | from_server_name |           from_server_version            | from_tps | to_server_name |            to_server_version             | to_tps | percent_change |
|---------------------|------------------|------------------------------------------|----------|----------------|------------------------------------------|--------|----------------|
| tpcc-scale-factor-1 | dolt             | 003f70ef15f7b23464d17a696552a3ae6fddff56 | 45.01    | dolt           | 4c1fbc805061a9681304251af50ce024697282d3 | 43.34  | -3.71          |

elianddb committed:
seed identity session vars from dolt config at connection time

elianddb committed:
use CommitIdentity at all commit identity call sites

elianddb committed:
rename commit identity types and refactor CommitSigner to interface

elianddb committed:
require non-nil CommitMeta in newCommitForValue

elianddb committed:
inline gpg.Signer.Sign and remove gpgSigner adapter

elianddb committed:
move remote connection identity check into BuildConnectionStringQueryist

elianddb committed:
remove remote engine skip from cherry-pick test

elianddb committed:
fix remaining author/committer call sites missed in initial pass

elianddb committed:
enginetests: add author/committer identity coverage for history, diff, and merge tables

elianddb committed:
bats: add author/committer identity integration tests

Tests cover: dolt commit with separate author/committer env vars, dolt show
displaying author vs committer, session variable identity in DOLT_COMMIT, and
distinct author/committer columns in dolt_diff and dolt_column_diff.

elianddb committed:
compat: add author/committer identity compatibility tests

coffeegoddd commented:
@elianddb __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>190d38a</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>190d38a</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@coffeegoddd __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>57423aa</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>57423aa</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

elianddb committed:
harden author/committer identity handling on the dolt client

Make dolt commit fail early via env.GetNameAndEmail when user.name or
user.email is missing so handleCommitErr still prints the standard
"Could not determine user.name" message. Seed identity session
variables in a single SHOW VARIABLES plus batched SET round trip from
InitCommitIdentSessionConfig, keeping older servers compatible. Cache
the resolved time on CommitDate so repeated reads return the same
value and decouple author from committer in resolveDates. Add
commit_order and author column skips to the server_cli_compat suite
and update ci, sql, sql-server, and database_provider tests to match
the new schema and local-loopback attribution rule.

coffeegoddd commented:
@elianddb __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>cc1d083</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>cc1d083</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@coffeegoddd __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>3901981</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>3901981</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@elianddb __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>3c91863</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>3c91863</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@coffeegoddd __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>230b96f</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>230b96f</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@elianddb __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>cc8704b</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>cc8704b</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@coffeegoddd __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>d075e64</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>d075e64</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@elianddb __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>b9258ac</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>b9258ac</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@coffeegoddd __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>6a3519d</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>6a3519d</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@elianddb __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>5d3f0d3</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>5d3f0d3</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@coffeegoddd __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>e5afe3b</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>e5afe3b</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@elianddb __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>f8b57d3</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>f8b57d3</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@coffeegoddd __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>1f18683</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>1f18683</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@elianddb __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>c8bd6d7</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>c8bd6d7</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@elianddb __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>7819c8d</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>7819c8d</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

macneale4 commented on .github/workflows/ci-compatibility-tests.yaml:
```diff
@@ -52,3 +52,4 @@ jobs:
         working-directory: ./integration-tests/compatibility
         env:
           DOLT_USE_ADAPTIVE_ENCODING: ${{ matrix.adaptive_encoding }}
+          BATS_PRINT_OUTPUT_ON_FAILURE: "1"
```
why? I thought this was the default

elianddb replied:
Not default, but it makes sense to pass as a flag down below more I think? I've removed it.

macneale4 commented on go/cmd/dolt/commands/commit.go:
```diff
@@ -235,20 +242,16 @@ func constructParametrizedDoltCommitQuery(msg string, apr *argparser.ArgParseRes
+		param = true
+		writeToBuffer("?")
+		author, _ := apr.GetValue(cli.AuthorParam)
+		params = append(params, author)
```
I'm missing something - is a lower level call using eng.GetNameAndEmail?

elianddb replied:
I looked into it a bit further, this is explicitly set to error out on the CLI end. I'm going to revert this as it was, but note how the `--author` flag is set. It specifically makes use of config values over the wire on the CLI.

macneale4 commented on integration-tests/bats/show.bats:
```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env bats
 load $BATS_TEST_DIRNAME/helper/common.bash
+# Tests in this file use dolt show to read raw NBS storage, which requires direct local
+# access to the data store and is not available in remote-engine mode.
```
why did you change this and remove the file form local-remote.bash? Is there a test that is run in remote mode?

elianddb replied:
I'll revert this one since it only added a little bit of coverage in the end.

macneale4 commented on integration-tests/bats/system-tables.bats:
```diff
@@ -1220,3 +1278,46 @@ SQL
+  [[ "$output" =~ ,email, ]] || false
+}
+
+assert_dolt_log() {
```
I don't see how the function name related to what it actually checks

elianddb replied:
I replaced the function name to specify its checking for an author and committer.

macneale4 commented on integration-tests/compatibility/test_files/bats/helper/server.bash:
Is this just a copy of the server helper in integrations-tests/bats/helper?

If so, perhaps symlink or something else. duplicate code is lame

elianddb replied:
not all of it just the definePORT part, these specifically rely on the NEW and OLD binaries. I've still added the ability to load those external recipes though.

macneale4 commented on integration-tests/compatibility/test_files/bats/server_cli_compat.bats:
I need to look at the existing state of backwards compatibility testing. Is all of this required?

elianddb replied:
For reading the `dolt_log` yes, since if the schema changes (which has happened as you can see with the skips) it breaks backwards compatibility for older clients cause of the expected indices.

macneale4 commented on integration-tests/bats/commit.bats:
```diff
@@ -191,6 +191,8 @@ SQL
 
 @test "commit: dolt commit works correctly with multiple branches" {
+    # dolt checkout is not supported in remote-engine mode.
+    skip_if_remote
```
The fix for this is to pass the --branch option. So, for example:

```
dolt branch branch2
dolt --branch branch2 sql -q "CREATE table t1 (pk int primary key);"
dolt --branch branch2 add t1
run dolt --branch branch2 commit -m "adding table t1 on branch1"
```

macneale4 commented on go/libraries/doltcore/sqle/dsess/transactions.go:
```diff
@@ -811,10 +811,17 @@ func (tx *DoltTransaction) ClearSavepoint(name string) bool {
 func (tx DoltTransaction) WorkingSetMeta(ctx *sql.Context) *datas.WorkingSetMeta {
-	sess := DSessFromSess(ctx.Session)
+	name, email, err := ResolveNameEmail(ctx, DoltCommitterName, DoltCommitterEmail)
+	if err != nil {
```
name and email will be nil when there is an error, yes? We swallow the error and convert to a warning, but I'm not sure what's going to happen downstream with nil name and emails. Correct me if I'm wrong.

elianddb replied:
You've got a point, I think the best bet is lifting this error out of the function, and passing the user and email as arguments instead, since that's what we really want. It's more of a constructor.

macneale4 commented on go/libraries/doltcore/sqle/dtables/branches_table.go:
```diff
@@ -243,6 +243,11 @@ func (bt *BranchesTable) Schema(ctx *sql.Context) sql.Schema {
 		columns = append(columns, &sql.Column{Name: "branch", Type: types.Text, Source: bt.tableName, PrimaryKey: false, Nullable: true})
 		columns = append(columns, &sql.Column{Name: "dirty", Type: types.Boolean, Source: bt.tableName, PrimaryKey: false, Nullable: true})
 	}
+	columns = append(columns,
```
Seems like the only tests impacted by this are the workbench tests. Seems to indicate some testing gaps in testing the `dolt_branches` table.

elianddb replied:
I've added some coverage to guard this in the future.

macneale4 commented on go/libraries/doltcore/sqle/enginetest/dolt_engine_test.go:
```diff
@@ -289,8 +289,8 @@ func TestSingleScriptPrepared(t *testing.T) {
-					sql.Row{0, 1, "itt2nrlkbl7jis4gt9aov2l32ctt08th", "billy bob", time.Date(1970, time.January, 1, 19, 0, 0, 0, time.Local)},
-					sql.Row{2, 3, "itt2nrlkbl7jis4gt9aov2l32ctt08th", "billy bob", time.Date(1970, time.January, 1, 19, 0, 0, 0, time.Local)},
+					sql.Row{0, 1, "itt2nrlkbl7jis4gt9aov2l32ctt08th", "root", time.Date(1970, time.January, 1, 19, 0, 0, 0, time.Local)},
+					sql.Row{2, 3, "itt2nrlkbl7jis4gt9aov2l32ctt08th", "root", time.Date(1970, time.January, 1, 19, 0, 0, 0, time.Local)},
```
This doesn't smell right. why are we returning root now when we were returning "billy bob" before. This screams of breaking user expectations.

elianddb replied:
This is a completely different hash in main, and also root. This is a single script test, I'll undo this change. But this has been broken

macneale4 commented on go/libraries/doltcore/sqle/enginetest/dolt_queries.go:
```diff
@@ -2683,6 +2683,32 @@ WHERE z IN (
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:    "SELECT committer FROM dolt_history_dolt_procedures WHERE name = 'p1'",
+				Expected: []sql.Row{{"root <root@localhost>"}},
```
Why isn't the committer test@example.com in this case? I thought when there wasn't a committer, we fell back to the author.

elianddb replied:
I'm changing this so if the committer is unset, the `--author` flag will affect both the committer and author.

macneale4 commented on go/cmd/dolt/commands/engine/sqlengine.go:
```diff
@@ -595,6 +596,96 @@ func doltSessionFactory(
+// DOLT_COMMITTER_* environment variable overrides the matching field. Must run after
+// [sql.SessionCommandBegin] so the session is ready to accept SET statements. Silently skips
+// variables not recognised by |queryist|, so older servers remain usable.
+func InitCommitIdentSessionConfig(queryist cli.Queryist, sqlCtx *sql.Context, name, email string) error {
```
rename to be very clear that this is intended for the CLI only.

elianddb replied:
I've renamed this to mention Client to make things clear.

macneale4 commented on go/libraries/doltcore/sqle/enginetest/dolt_queries_cherry_pick.go:
```diff
@@ -683,7 +683,7 @@ var DoltCherryPickTests = []queries.ScriptTest{
 			},
 			{
-				Query:    "select committer, email, message, date from dolt_log limit 1;",
+				Query:    "select author, author_email, message, author_date from dolt_log limit 1;",
```
I would expect additional tests in dolt_queries_cherry_pick.go rather than updating existing tests. Ideally all existing tests should be left alone so that we can verify we aren't changing behavior.

elianddb replied:
This is a correctness issue for this one. The author is taken from the old commit, so I think this change is necessary. But I will add additional tests.

elianddb replied:
Modifying to an additvie change.

macneale4 commented on go/libraries/doltcore/sqle/enginetest/dolt_queries_merge.go:
```diff
@@ -393,7 +393,7 @@ var MergeScripts = []queries.ScriptTest{
 			},
 			{
-				Query:    "select message from dolt_log order by date DESC LIMIT 1;",
+				Query:    "select message from dolt_log order by author_date DESC LIMIT 1;",
```
I'll repeat my assertion that existing tests should be touched _as little as possible_. This change should not be required. Same goes for several other tests in this file.

elianddb replied:
`--date` was modified to set committer as well now

elianddb commented on go/libraries/doltcore/sqle/enginetest/dolt_queries.go:
```diff
@@ -5867,7 +5939,7 @@ var DoltTagTestScripts = []queries.ScriptTest{
 			{
 				Query:    "SELECT tag_name, IF(CHAR_LENGTH(tag_hash) < 0, NULL, 'not null'), tagger, email, IF(date IS NULL, NULL, 'not null'), message from dolt_tags",
-				Expected: []sql.Row{{"v1", "not null", "billy bob", "bigbillieb@fake.horse", "not null", ""}},
+				Expected: []sql.Row{{"v1", "not null", "root", "root@localhost", "not null", ""}},
```
I'm keeping this change for consistency, only making an exception for the initialization of the repository.

elianddb committed:
amend committer related tests using --author flag and fix bats compatibility testing to use existing helpers

elianddb committed:
Merge remote-tracking branch 'origin/main' into elian/1374

elianddb committed:
amend dolt_branches integrations to check schema too

elianddb committed:
amend bats and workbench mysql-client integrations

coffeegoddd commented:
@elianddb __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>c6e1eb3</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>c6e1eb3</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@elianddb __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>82ef18d</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>82ef18d</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

elianddb commented on go/cmd/dolt/commands/commit.go:
```diff
@@ -235,20 +239,15 @@ func constructParametrizedDoltCommitQuery(msg string, apr *argparser.ArgParseRes
+		writeToBuffer("?")
+		author, _ := apr.GetValue(cli.AuthorParam)
+		params = append(params, author)
+	} else if _, _, err := env.GetNameAndEmail(cliCtx.Config()); err != nil {
```
Otherwise CLI will stop the session variable from working, but we still want the error.

coffeegoddd commented:
@elianddb __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>cea3aca</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>cea3aca</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@elianddb __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>b261fe4</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>b261fe4</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

coffeegoddd commented:
@elianddb __DOLT__
<table>
<TR><TH>comparing_percentages</TH>
</TR>
<TR><TD>100.000000 to 100.000000</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>result</TH>
<TH>total</TH>
</TR>
<TR><TD>4c1fbc8</TD>
<TD>ok</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>version</TH>
<TH>total_tests</TH>
</TR>
<TR><TD>4c1fbc8</TD>
<TD>5937471</TD>
</TR>
</table>
<table>
<TR><TH>correctness_percentage</TH>
</TR>
<TR><TD>100.0</TD>
</TR>
</table>

macneale4 commented on integration-tests/compatibility/test_files/bats/2_0_breaking/2_0_breaking.bats:
```diff
@@ -6,9 +6,8 @@
 # containing TEXT and BLOB columns, then verify that an older dolt client (pre-2.0) fails
 # with "table has unknown fields" when trying to read them.
 
-dolt config --global --add metrics.disabled true > /dev/null 2>&1
```
What?? I can't imagine why we'd do this. We don't want our bats tests to show up in metrics.

elianddb replied:
set in `common.bash` original helpers

macneale4 commented on integration-tests/compatibility/test_files/bats/bidirectional/bidirectional_compat.bats:
```diff
@@ -10,9 +10,8 @@
 # Note that the setup files in |helper| can't be used here because they use relative paths and we
 # are in a subdirectory. So the key pieces of functionality are duplicated in this file.
 
-dolt config --global --add metrics.disabled true > /dev/null 2>&1
```
again - huh? Is this set elsewhere and this is just no needed anymore?

elianddb replied:
set in `common.bash` original helpers

macneale4 reviewed (approved):
Looks good - but I do have one question about disabling metrics. convince me that's right before you merge
